helper/compiler_factory.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import tarfile
+import os
+import re
+import importlib
+from urllib.request import urlretrieve
+from pathlib import Path
+from helper.cd import cd
+from shutil import which
+
+
+class CompilerFactory(object):
+    def __init__(self, toolchain_url, toolchain_extractpath):
+        self.toolchain_url = toolchain_url
+        self.toolchain_extractpath = toolchain_extractpath
+
+    def getCompiler(self):
+        if self.toolchain_url.find('http') != -1 or self.toolchain_url.find('ftp') != -1:
+            extracted_tar = self._downloadToolchain()
+            return self._fetchCompiler(extracted_tar)
+        elif self.toolchain_url.find('g++') != -1:
```

Comment:
you can replace this with one if match this and match that: fetch_system(toolchain_url)